### PR TITLE
Issue #3958: Modified GenericWhitespaceCheckTest.java and moved its i…

### PR DESCRIPTION
**Pull request by subkrish**
Issue #3958

### CODE REVIEW

**Comment by Vladlis** [comment]
All commits need to be squashed in one.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceList.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.puppycrawl.tools.checkstyle.checks.whitespace.genericwhitespace;
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceList.java]: this file also need to be moved by "git mv" to preserve history.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputMethodReferences3.java [hunk 2]
```diff
@@ -1,6 +1,7 @@
 //Compilable with Java8
 package com.puppycrawl.tools.checkstyle.checks.whitespace;
 import java.util.function.Supplier;
+
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputMethodReferences3.java]: please remove this change

**Comment by romani** [comment]
- please keep reply to each point as - "done"
- please rebase on latest code , in your diff change for "config_coding.xml" is not expected

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheckTest.java [hunk 3]
```diff
@@ -105,11 +105,11 @@ public void testDefault() throws Exception {
...
         };
-        verify(checkConfig, getPath("InputGenericWhitespace.java"), expected);
+        verify(checkConfig, getPath("InputGenericWhitespaceDefault.java"), expected);
     }
 
     @Test
-    public void testGh47() throws Exception {
+    public void testList() throws Exception {
         final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;
         verify(checkConfig, getPath("InputGh47.java"), expected);
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheckTest.java]: Shouldn't this file name change too?

File: src/xdocs/config_coding.xml [hunk 4]
```diff
@@ -50,12 +50,31 @@ return new int[] { 0 };
       </subsection>
 
       <subsection name="Examples">
-        <p>
```
Comment (rnveach) [inline src/xdocs/config_coding.xml]: no change is expected in this file like romani said.

**Comment by subkrish** [comment]
I have made the changes requested.

**Comment by rnveach** [comment]
@subkrish There is still a conflict in `InputGenericWhitespaceMethodRef.java`. Make sure you are using the latest master.

**Comment by subkrish** [comment]
Conflict resolved. Done.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceMethodRef2.java [hunk 5]
```diff
@@ -1,10 +1,10 @@
-package com.puppycrawl.tools.checkstyle.checks.whitespace;
+//Compilable with Java8
```
Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceMethodRef2.java]: Look at diff. This line wasn't here before, you added it when resolving the conflicts. We just recently removed this comment from everywhere.
Please remove it.

**Comment by subkrish** [comment]
Sorry for that. There was an issue I had with rebasing so saved a copy of the changes I made. I have removed that line now. Done.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceList.java [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 import java.util.List;
 
-public class InputGh47
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceList.java]: please change package of this class to new

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputGenericWhitespaceDefault.java [hunk 7]
```diff
@@ -7,7 +7,7 @@
 import java.util.Map;
 import java.util.concurrent.Callable;
 import java.util.Collections;
-class InputGenericWhitespace implements Comparable<InputGenericWhitespace>, Serializable
```
Comment (romani) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputGenericWhitespaceDefault.java]: please keep name of top class as is, it is not required.

File: src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/GenericWhitespaceTest.java [hunk 8]
```diff
@@ -104,7 +104,7 @@ public void genericWhitespaceTest() throws Exception {
             "63:60: " + getCheckMessage(checkConfig.getMessages(), msgFollowed, ">"),
         };
 
-        final String filePath = getPath("InputGenericWhitespace.java");
+        final String filePath = getPath("InputGenericWhitespaceDefault.java");
```
Comment (romani) [inline src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/GenericWhitespaceTest.java]: changes in ITs are not expected

**Comment by subkrish** [comment]
For the InputGenericWhitespaceList file, after changing it from InputGh47 it is not preserving history even after doing 'git mv' also. I have tried manually but it still reflects as deleting the file and then creating it again. Along with that I have reverted the changes that were made in the 'it' folder.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceList.java (hunk 6 above)
Comment (subkrish) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/genericwhitespace/InputGenericWhitespaceList.java]: done.

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputGenericWhitespaceDefault.java (hunk 7 above)
Comment (subkrish) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputGenericWhitespaceDefault.java]: done.

File: src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/GenericWhitespaceTest.java (hunk 8 above)
Comment (subkrish) [inline src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/GenericWhitespaceTest.java]: done.

**Comment by romani** [comment]
> t is not preserving history even after doing 'git mv' also. 

did you try to "git mv oldpath new path" and after that "git mv oldname newname" ? (2 commands )

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputGenericWhitespace.java [hunk 9]
```diff
@@ -62,4 +62,4 @@ public int getConstructor(Class<?>... parameterTypes)
 
     public static class IntEnumValueType3<E extends Enum<E>  & IntEnum> { // warn
     }
-}
+}
```
Comment (romani) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputGenericWhitespace.java]: please revert changes in this file.

**Comment by subkrish** [comment]
As suggested I have done 'git mv' in 2 steps. But it is still recognizing it as an old file deletion and new file creation. I am not too sure but I think it has something to do with the amount of the content present in the file that has been changed. After moving the file using 'git mv', I had to change the name of the class and add an extension to the package which is causing git to see it as large change because the content of the file is small.

**Comment by romani** [comment]
ok , very strange, I am fine with changes.

@rnveach , please do final review.

**Pull request merged**